apps/lockfile-explorer/src/graph/pnpmfileRunnerWorkerThread.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { parentPort, workerData } from 'node:worker_threads';
+import * as path from 'node:path';
+import * as fs from 'node:fs';
+import type { IPackageJson } from '@rushstack/node-core-library';
+
+import type { IPnpmfileModule, IReadPackageContext } from './IPnpmfileModule';
+
+export interface IRequestMessage {
+  id: number;
+  packageJson: IPackageJson;
+  packageJsonFullPath: string;
+}
+
+export interface IResponseMessageLog {
+  kind: 'log';
+  id: number;
+  log: string;
+}
+export interface IResponseMessageError {
+  kind: 'error';
+  id: number;
+  error: string;
+}
+export interface IResponseMessageReturn {
+  kind: 'return';
+  id: number;
+  result?: unknown;
+}
+export type ResponseMessage = IResponseMessageLog | IResponseMessageError | IResponseMessageReturn;
+
+// debugger;
+
+const { pnpmfilePath } = workerData;
+const resolvedPath: string = path.resolve(pnpmfilePath);
```

Comment:
Is this ever not absolute?

Reply:
It is absolute, but only because in the current implementation, an absolute path is used by the caller of the class constructor in the other thread. It is a very brittle assumption.